cmd/main.go

```diff
@@ -0,0 +1,211 @@
+
+		if out, err := cmd.CombinedOutput(); err != nil {
+			log.Println(string(out))
+			return err
+		}
+
+		return nil
+	default:
+		return errorPlatformNotSupported
+	}
+}
+
+func downloadLatestReleaseAsset() string {
+	release, error := getLatestRelease()
+
+	if error != nil {
+		panic(error)
+	}
+
+	resp, _ := http.Get(*release.asset.BrowserDownloadURL)
+	defer resp.Body.Close()
+
+	out, _ := os.Create(*release.asset.Name)
+	defer out.Close()
+
+	log.Printf("Downloading latest Flowser release %s", release.version)
+
+	_, err := io.Copy(out, resp.Body)
+	if err != nil {
+		panic("Install failed")
+	}
+
+	return out.Name()
+}
+
+type FlowserRelease struct {
+	version string
+	asset   github.ReleaseAsset
+}
+
+func getLatestRelease() (FlowserRelease, error) {
+	client := github.NewClient(nil)
+	release, _, err := client.Repositories.GetLatestRelease(context.Background(), "onflowser", "flowser")
+	if err != nil {
+		log.Fatal(err)
+	}
+	latestVersion := strings.Replace(*release.TagName, "v", "", 1)
+	targetAssetName, err := getAssetName(latestVersion)
+	if err != nil {
+		return FlowserRelease{}, err
+	}
+	for _, asset := range release.Assets {
+		if *asset.Name == targetAssetName {
+			return FlowserRelease{
+				version: latestVersion,
+				asset:   asset,
+			}, nil
+		}
+	}
+	return FlowserRelease{}, errors.New("no asset found")
+}
+
+func getAssetName(version string) (string, error) {
+	isArm := strings.HasPrefix(runtime.GOARCH, "arm")
+	switch runtime.GOOS {
+	case "darwin":
+		if isArm {
+			return fmt.Sprintf("Flowser-%s-arm64-mac.zip", version), nil
+		} else {
+			return fmt.Sprintf("Flowser-%s-mac.zip", version), nil
+		}
+	case "windows":
+		return fmt.Sprintf("Flowser-%s-win.zip", version), nil
+	default:
+		return "", errorPlatformNotSupported
+	}
+}
+
+func getInstallDir() (string, error) {
+	switch runtime.GOOS {
+	case "darwin":
+		return "/Applications/Flowser.app", nil
+	case "windows":
+		// TODO: Search in common install directories
+		// https://superuser.com/questions/1327037/what-choices-do-i-have-about-where-to-install-software-on-windows-10
+		user, err := user.Current()
+		if err != nil {
+			return "", err
+		}
+		return fmt.Sprintf("%s\\AppData\\Local\\Programs\\Flowser", user.HomeDir), nil
+	default:
+		return "", errorPlatformNotSupported
+	}
+}
+
+func getExecutableFile() (string, error) {
+	installDir, err := getInstallDir()
+	if err != nil {
+		return "", err
+	}
+	switch runtime.GOOS {
+	case "darwin":
+		return path.Join(installDir, "Contents/MacOS/Flowser"), nil
+	case "windows":
+		return path.Join(installDir, "Flowser.exe"), nil
+	default:
+		return "", errorPlatformNotSupported
+	}
+}
+
+func Run(location string) error {
+	execFilePath, err := getExecutableFile()
+	if err != nil {
+		return err
+	}
+	cmd := exec.Command(execFilePath, fmt.Sprintf("--project-path=%s", location))
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func main() {
```

Comment:
This should be moved to cmd/main, also not sure if it's needed. It feels to me you need to write a test for what you are trying to validate here.